sale_order_line_description/models/sale.py

```diff
@@ -19,34 +19,25 @@
 #
 #
 
-from openerp import models, api
+from odoo import models, api
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
     @api.multi
-    def product_id_change(
-            self, pricelist, product, qty=0, uom=False, qty_uos=0,
-            uos=False, name='', partner_id=False, lang=False,
-            update_tax=True, date_order=False, packaging=False,
-            fiscal_position=False, flag=False
-    ):
-        res = super(SaleOrderLine, self).product_id_change(
-            pricelist=pricelist, product=product, qty=qty, uom=uom,
-            qty_uos=qty_uos, uos=uos, name=name,
-            partner_id=partner_id, lang=lang, update_tax=update_tax,
-            date_order=date_order, packaging=packaging,
-            fiscal_position=fiscal_position, flag=flag)
-        if product:
-            product_obj = self.env['product.product']
+    @api.onchange('product_id')
+    def product_id_change(self):
+        res = super(SaleOrderLine, self).product_id_change()
+        if self.product_id:
             if self.user_has_groups(
                     'sale_order_line_description.'
                     'group_use_product_description_per_so_line'):
-                lang = self.env['res.partner'].browse(partner_id).lang
-                product = product_obj.with_context(lang=lang).browse(product)
+                product = self.product_id.with_context(
+                    lang=self.order_id.partner_id.lang,
+                )
                 if product.description_sale:
-                    if 'value' not in res:
-                        res['value'] = {}
-                    res['value']['name'] = product.description_sale
+                    vals = {'name': product.description_sale}
+                    self.update(vals)
```

Comment:
I tried on runbot but this seem do not work, take a cue from https://github.com/tafaRU/product-variant/blob/10.0-mig-sale_order_line_variant_description/sale_order_line_variant_description/models/sale_order_line.py

Reply:
I used self.update because it is the exact copy of how the overridden method works: https://github.com/odoo/odoo/blob/10.0/addons/sale/models/sale.py#L907

I have also recently added a unit test to check the behavior and it seems to work on runbot

Reply:
Ok thanks!